src/components/LinkChooser/LinkChooser.tsx

```diff
@@ -0,0 +1,218 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { Checkbox, CheckboxState } from "@components/Checkbox/Checkbox";
+import { mapToAriaProps } from "@components/Menu/Aria/helper";
+import IconDocument from "@foundation/Icon/Generated/IconDocument";
+import IconDocumentLibrary from "@foundation/Icon/Generated/IconDocumentLibrary";
+import IconExternalLink from "@foundation/Icon/Generated/IconExternalLink";
+import IconLink from "@foundation/Icon/Generated/IconLink";
+import IconTemplate from "@foundation/Icon/Generated/IconTemplate";
+import { useComboBox } from "@react-aria/combobox";
+import { DismissButton } from "@react-aria/overlays";
+import { useComboBoxState } from "@react-stately/combobox";
+import { useMachine } from "@xstate/react";
+import { AnimatePresence, motion } from "framer-motion";
+import React, { FC, Key, ReactElement, useCallback, useEffect, useMemo, useRef } from "react";
+import { Popover } from "./Popover";
+import { SearchInput } from "./SearchInput";
+import { SearchResultsList } from "./SearchResultSection";
+import { SectionActionMenu } from "./SectionActionMenu";
+import { linkChooserMachine, LinkChooserState } from "./state/machine";
+import { LinkChooserProps } from "./types";
+import * as SearchRepository from "./repositories";
+import { createCustomLink, isLoaded, queryMatchesSelection } from "./utils/helpers";
+
+export enum IconLabel {
+    Document = "DOCUMENT",
+    Library = "LIBRARY",
+    Link = "LINK",
+    External = "EXTERNAL",
+    Template = "TEMPLATE",
+}
+
+export const ICON_OPTIONS: Record<IconLabel | string, ReactElement> = {
+    [IconLabel.Document]: <IconDocument />,
+    [IconLabel.Library]: <IconDocumentLibrary />,
+    [IconLabel.Link]: <IconLink />,
+    [IconLabel.External]: <IconExternalLink />,
+    [IconLabel.Template]: <IconTemplate />,
+};
+
+export const DEFAULT_ICON = IconLabel.Link;
+export const CUSTOM_LINK_ID = "custom-link";
+export const MAX_STORED_ITEMS = 5;
+export const QUERIES_STORAGE_KEY = "queries";
+
+export const LinkChooser: FC<LinkChooserProps> = ({
+    getGlobalByQuery = SearchRepository.getGlobalByQuery,
+    getTemplatesByQuery = SearchRepository.getTemplatesByQuery,
+    openPreview = window.open,
+    copyToClipboard = navigator.clipboard,
+    openInNewTab,
+    ariaLabel = "Menu",
+    label,
+    placeholder,
+    onOpenInNewTabChange,
+    onLinkChange,
+}) => {
+    const [{ context, matches, value }, send, service] = useMachine(
+        linkChooserMachine.withContext({
+            searchResults: [],
+            selectedResult: null,
+            query: "",
+            interruptedFetch: false,
+            copyToClipboard,
+            openPreview,
+            getGlobalByQuery,
+            getTemplatesByQuery,
+            onLinkChange,
+        }),
+    );
+
+    const handleClearClick = useCallback(() => {
+        state.setInputValue("");
+        state.setSelectedKey("");
+        send("CLEARING", { data: { query: "" } });
```

Comment:
since "CLEARING" is reusing an action that requires a defined `query` parameter, i have to send an empty string so that it properly works and clears the query